reports/aksversions/README.md

```diff
@@ -8,50 +8,60 @@ The scripts goal is to find every AKS cluster within a set of subscriptions, fin
 
 There are 2 scripts that make this report work:
 
-- `aks-versions.sh` - This script does the search and json build
+- `main.py` - This script searches Azure for AKS clusters and creates a JSON object of relevant information.
     - Searches for all subscriptions containining `SHAREDSERVICES` or `CFT` and addes the subscription Id into an array
     - Uses the list of subscriptions to then search for deployed AKS clusters and adds this information to another array
     - Checks the discovered AKS clusters for available updates
     - Builds an new json object containing all the relevant information so it can be added to a Cosmos container
 - `save-to-cosmos.py` - This script is only used to interact with Cosmos DB
     - Will remove all documents from the chosen container
-    - Will then add all new documents supplied to the script from the `aks-versions.sh` script
+    - Will then add all new documents supplied to the script from the `main.py` script
 
 ## Dockerfile
 
-The Dockerfile will build an image that contains the scripts and will run the `aks-versions.sh` script when launched.
+The Dockerfile will build an image that contains the scripts and will run the `main.py` script when launched.
 This container image will be run as a cronjob so that it runs on a schedule and when complete the pod will stop and eventually be removed.
 
 The Dockerfile does not have any effect on the report process and is simply a way to make this deployable to AKS.
 
 ## Local dev
 
-It is possible to run this script locally if you have:access to the Azure as you will need to:
+As this report utilises Python you will need to have it installed, development was carried out using `Python 3.11.7` so a version greater than `3.11.x` is recommened.
 
-- Log into Azure via the `azure cli` so the script can use your credentials/access to read the subscriptions and AKS cluster information.
+The main python script requires a service principal to run with the values supplied as environment variables:
 
-and also set the following environment variables:
+- AZURE_CLIENT_ID
+- AZURE_CLIENT_SECRET
+- AZURE_TENANT_ID
+
+The service principal used within AKS can be found as secrets in the `monitoring namespace` and there are documented methods to decrypt these kinds of [secrets](https://stackoverflow.com/questions/56909180/decoding-kubernetes-secret).
```

Comment:
we shouldn't be using service principals with fixed secrets in AKS ever.

Use a managed identity with workload identity.

Locally you should use azure cli credentials, and in docker locally for testing you can use this to not need additional credentials:
https://github.com/gsoft-inc/azure-cli-credentials-proxy